src/main/java/uk/ac/ebi/tsc/portal/api/team/service/TeamService.java

```diff
@@ -924,4 +924,15 @@ public TeamResource populateTeamMemberEmails(Team team, TeamResource teamResourc
 		}
 		return teamResource;
 	}
+
+	public Iterable checkAndSetMemberTeams(Collection<Team> teams, String username) {
+		return teams.parallelStream().map(team -> {
+			TeamResource teamResource = new TeamResource(team);
+			Account member = team.accountsBelongingToTeam.stream().filter(a -> a.getUsername().equals(username)).findAny().orElse(null);
+			if (member != null) {
+				teamResource.getMemberAccountEmails().add(member.email);
+			}
+			return teamResource;
+		}).collect(Collectors.toList());
+	}
```

Comment:
Looks like n+1 for me. Isn't there a separate call for `accountsBelongingToTeam` for each team now? I believe so.
Better option - have one single call for teams belonging to a given user and just merge results. This is actually what I wrote originally.
If you need to do that like that - then your only non n+ 1 option is JOIN FETCHING teams with users and then filtering.